### PR TITLE
Add 3D localization demo.

### PR DESCRIPTION
**Pull request by wohe**
Adds configuration files and launch files for 3D backpack pure
localization.

Moves most demos to a separate documentation page.

### CODE REVIEW

File: docs/source/demo.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    roslaunch cartographer_ros demo_backpack_2d_localization.launch \
+       map_filename:=${HOME}/Downloads/b2-2016-04-05-14-44-52.bag.pbstream \
+       bag_filename:=${HOME}/Downloads/b2-2016-04-27-12-31-41.bag
+
+    # Pure localization demo in 3D: We use 2 different 3D bags from the Deutsche
+    # Museum. The first one is used to generate the map, the second to run
+    # pure localization.
+    wget -P ~/Downloads https://storage.googleapis.com/cartographer-public-data/bags/backpack_3d/b3-2016-04-05-13-54-42.bag
+    wget -P ~/Downloads https://storage.googleapis.com/cartographer-public-data/bags/backpack_3d/b3-2016-04-05-15-52-20.bag
+    # Generate the map: Run the next command, wait until cartographer_offline_node finishes.
```
Comment (ojura) [inline docs/source/demo.rst]: Assets are not written at the end of offline node anymore, right? Now you get a .pbstream and need to run assets writer. See #577.

Comment (wohe) [reply docs/source/demo.rst]: Correct. This does not write assets, but it writes the `.pbstream` which is all we need. You do not have to run the assets writer for this demo.

Comment (ojura) [reply docs/source/demo.rst]: Okay, but that means that the "Generate the map" part of the command is not accurate.

Comment (gaschler) [reply docs/source/demo.rst]: We localize in a map, and that map is in a pbstream file (not an image or other asset). Perhaps the term "map" is overloaded here but I think this is fine.

Comment (ojura) [reply docs/source/demo.rst]: Oh right, sorry, I missed the context.

**Pull request merged**